DataBiosphere/leonardo
# [IA-4364] Updating docker images to python 3.10

__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4364
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-3994
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4271
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4304
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4356

 ## Dependencies
[Terra docker PR that generated the images](https://github.com/DataBiosphere/terra-docker/pull/434/files)

## Summary of changes



### What

- Updates the base image from` tf-gpu.2-7` to `tf-gpu.2-11.py310` to use python 3.10
- Removes references to gatk-otvf image
- Enables toc2 jupyter extension to allow file downloads in different formats
- cromshell 2.0 fix

### Why

- Python 3.7 is EOL (no more bug patches) at the end of June
- Intel decided to stop development of OVTF and will no longer be maintained
- Users want to save their notebooks as HTML or other formats

## Testing these changes

### What to test

Choose an image:
- create a jupyter notebook with their image and default settings
- run through the T101 notebook to make sure basic functionality works (can be in any workspace)
- https://app.terra.bio/#workspaces/help-terra/T101-Notebooks-Quickstart/analysis/launch/Analyze-data-from-table.ipynb
- copy the workspace of their featured notebook
    - create a Jupyter notebook with their image
    - run all cells for their respective notebook(s)
- try downloading their notebook as a non-IPYNB file

* R/Bioconductor
    * us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor:2.2.1
    * https://app.terra.bio/#workspaces/help-gatk/Bioconductor/analysis/launch/SingleCellExperiment.ipynb
    * https://app.terra.bio/#workspaces/help-gatk/Bioconductor/analysis/launch/edgeR.ipynb
* Hail
    * us.gcr.io/broad-dsp-gcr-public/terra-jupyter-hail:1.1.1
    * https://app.terra.bio/#workspaces/help-gatk/Hail-Notebook-Tutorials/analysis/launch/01-genome-wide-association-study.ipynb 
    * https://app.terra.bio/#workspaces/featured-workspaces-hca/Intro-to-HCA-data-on-Terra/analysis/launch/07-matrixtable.ipynb
* Python
    * us.gcr.io/broad-dsp-gcr-public/terra-jupyter-python:1.1.1
    * https://app.terra.bio/#workspaces/featured-workspaces-hca/Intro-to-HCA-data-on-Terra/analysis/launch/Pegasus.ipynb 
* Default
    * us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk:2.3.1
    * https://app.terra.bio/#workspaces/pathogen-genomic-surveillance/COVID-19/analyses
    * https://app.terra.bio/#workspaces/broad-firecloud-dsde/Hyperparameter%20Tuning%20with%20Optuna/analyses 

### Who tested and where

  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
